truffle.js

```diff
@@ -1,6 +1,12 @@
 var HDWalletProvider = require("truffle-hdwallet-provider");
 var mnemonic = "digital unknown jealous mother legal hedgehog save glory december universe spread figure custom found six"
 
+const secret_mnemonic_file = "./secret_mnemonic"
+const fs=require('fs')
+let secret_mnemonic
+if (fs.existsSync(secret_mnemonic_file)) {
```

Comment:
move this into the "xdai_poa_mainnet" block, below, so we don't need it unless we try to contact xdai